buildenv/jenkins/common/variables-functions

```diff
@@ -845,27 +856,31 @@ def get_build_specs_by_release(version, specs) {
  * and test. Returns a map of releases by specs.
  */
 def get_specs(SUPPORTED_SPECS) {
-    def RELEASES = []
-    def PLATFORMS = []
     def SPECS = [:]
 
-    // get releases 
-    params.each { key, value  ->
-        if ((key.indexOf('Java') != -1) && (value == true)) {
-            RELEASES.add(key.substring(4))
+    if (!RELEASES) {
+        // get releases
+        params.each { key, value  ->
+            if ((key.indexOf('Java') != -1) && (value == true)) {
+                RELEASES.add(key.substring(4))
+            }
         }
     }
+    echo "RELEASES:'${RELEASES}'"
 
-    // get specs
-    if (!params.PLATFORMS || (params.PLATFORMS.trim() == 'all')) {
-        // build and test all supported specs
-        PLATFORMS.addAll(SUPPORTED_SPECS.keySet())
-    } else {
-        PLATFORMS.addAll(params.PLATFORMS.split(","))
+    if (!PLATFORMS) {
+        // get specs
+        if (!params.PLATFORMS || (params.PLATFORMS.trim() == 'all')) {
+            // build and test all supported specs
+            PLATFORMS.addAll(SUPPORTED_SPECS.keySet())
+        } else {
+            PLATFORMS.addAll(params.PLATFORMS.split(","))
```

Comment:
PLATFORMS is expected to be an array. Build parameters are also env variables, thus if PLATFORMS is set as build parameter, this code is never executed and we want the build parameter PLATFORMS (comma delimited string) to be converted to array.
Is it declared anywhere `def PLATFORMS = []`?

Reply:
Think I fixed it
https://ci.eclipse.org/openj9/view/Sandbox/job/Adam-Pipeline-Build-Test-All/43/console
```
...
PLATFORMS:'x86-64_linux,ppc64le_linux'
...
11:43:50  PLATFORMS:'[x86-64_linux, ppc64le_linux]'
```
https://ci.eclipse.org/openj9/job/PullRequest-OpenJ9/191/console
```
...
PLATFORMS:'null'
...
11:46:16  PLATFORMS:'[x86-64_linux, ppc64le_linux]'
```